presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp

```diff
@@ -234,6 +258,133 @@ TEST_P(HttpTestSuite, timedOutRequests) {
   wrapper.stop();
 }
 
+TEST_P(HttpTestSuite, httpConnectTimeout) {
+  const bool useHttps = GetParam();
+  if (!useHttps) {
+    // Cannot test connect timeout for HTTP as HTTP connection establishment is
+    // handled in kernel.
+    return;
+  }
+  auto memoryPool = defaultMemoryManager().addLeafPool("httpTimeouts");
+
+  auto server = getServer(useHttps, 1);
+
+  auto request1 = std::make_shared<AsyncMsgRequestState>();
+  auto request2 = std::make_shared<AsyncMsgRequestState>();
+
+  server->registerGet("/async/msg1", asyncMsg(request1));
+  server->registerGet("/async/msg2", asyncMsg(request2));
+
+  HttpServerWrapper wrapper(std::move(server));
+  auto serverAddress = wrapper.start().get();
+
+  HttpClientFactory clientFactory;
+  auto client1 = clientFactory.newClient(
+      serverAddress,
+      std::chrono::milliseconds(5'000),
```

Comment:
Let's set the request timeout to a large value to avoid the request timeout in this test? Thanks!